### PR TITLE
Operator overloading on RHS

### PR DESCRIPTION
**Pull request by AndreyAPopov**
This is a test as of now.

### CODE REVIEW

**Comment by AndreyAPopov** [comment]
So here is how all this works in matlab:

```
>> model = otp.brusselator.presets.Canonical;
>> firstCompRHS = model.RHS(1, :);
>> secondCompRHS = model.RHS(2, :);
>> reverseRHS = [secondCompRHS; firstCompRHS];
>> model.RHS.F(0, model.Y0)
ans =
    -6
     5
>> reverseRHS.F(0, model.Y0)
ans =
     5
    -6
```

**Comment by Steven-Roberts** [comment]
This should be ready to merge after some testing

File: src/+otp/RHS.m [hunk 1]
```diff
@@ -48,5 +118,99 @@
...
+            
+            % JPattern is problematic to compute for division operators due to
+            % singular patterns
+            
+            % Mass matrices introduce several difficulties. When singular, it
+            % makes it infeasible to update InitialSlope, and therefore, it is
+            % always unset. To avoid issues with two RHS' having different mass
+            % matrices, only the primary RHS is used.
+            
+            newRHS = otp.RHS(f, ...
```
Comment (AndreyAPopov) [inline src/+otp/RHS.m:180]: Mass matrices are a big difficulty right now it seems like.

**Comment by AndreyAPopov** [comment]
I feel like what we need now is to setup some pretty serious edge-case tests to exhaustively test every possible scenario

**Comment by AndreyAPopov** [comment]
As it is now, there is effectively no guarantees that this will work as intended

File: src/+otp/RHS.m (hunk 1 above)
Comment (Steven-Roberts) [reply src/+otp/RHS.m:180]: I don't think there is much more that can be done outside of warnings for incompatible mass matrices.

Comment (AndreyAPopov) [reply src/+otp/RHS.m:180]: I agree. That is fine

**Comment by Steven-Roberts** [comment]
I say we go ahead and merge this.  Clearly it needs additional testing, but proper testing for OTP will be a large undertaking in a separate PR.

**Comment by AndreyAPopov** [comment]
> I say we go ahead and merge this. Clearly it needs additional testing, but proper testing for OTP will be a large undertaking in a separate PR.

Fine. It is in a usable state. We will get more mileage out of testing by people actually using it.
